### PR TITLE
[ROCm] Enable 3D batch norms through MIOpen

### PR DESCRIPTION
**Pull request by iotamudelta**
Enable test for Caffe2

### CODE REVIEW

**Comment by iotamudelta** [comment]
CC: @lcskrishna

**Review by bddppq — COMMENTED** [review]
Thanks for adding the 3D support.
Caffe2 part looks good. For the PyTorch part, could you add a test case?

**Comment by iotamudelta** [comment]
@bddppq I am not sure I understand your request. Unlike Caffe2, PyTorch so far simply used the internal fallback batch norm. Now it uses MIOpen, the same test cases apply, simply using the new code path.

E.g., test_nn has:
```
test_BatchNorm3d (__main__.TestNN) ... ok
test_BatchNorm3d_3d_simple_average (__main__.TestNN) ... ok
test_BatchNorm3d_3d_simple_average_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_3d_simple_average_eval (__main__.TestNN) ... ok
test_BatchNorm3d_3d_simple_average_eval_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_eval (__main__.TestNN) ... ok
test_BatchNorm3d_eval_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_momentum (__main__.TestNN) ... ok
test_BatchNorm3d_momentum_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_momentum_eval (__main__.TestNN) ... ok
test_BatchNorm3d_momentum_eval_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_not_affine (__main__.TestNN) ... ok
test_BatchNorm3d_not_affine_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_not_affine_eval (__main__.TestNN) ... ok
test_BatchNorm3d_not_affine_eval_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_not_tracking_stats (__main__.TestNN) ... ok
test_BatchNorm3d_not_tracking_stats_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_not_tracking_stats_eval (__main__.TestNN) ... ok
test_BatchNorm3d_not_tracking_stats_eval_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_zero_batch (__main__.TestNN) ... ok
test_BatchNorm3d_zero_batch_cuda (__main__.TestNN) ... ok
test_BatchNorm3d_zero_batch_eval (__main__.TestNN) ... ok
test_BatchNorm3d_zero_batch_eval_cuda (__main__.TestNN) ... ok
```

**Comment by bddppq** [comment]
@iotamudelta Oh I didn't realize it was using the fallback.

**Review by bddppq — APPROVED** [review]
LGTM
